### ISSUE TITLE
timeout error comes up more and more  frequently during project CI

### ISSUE THREAD

**Issue by huangyuqi** [kubernetes/heapster#1349]
Now, the PR checks unsuccessfully more frequently in ` continuous-integration/travis-ci/pr `, with the error log:
```
Your test run exceeded 120 minutes. 
One possible solution is to split up your test run.
```
The reason is obvious, 
- the project of Heapster becomes bigger day by day. we need to run CI check `go test` parallelly.
- And in current version, the CI script run `go test` in all file path in the project, including `vendor`; we need to find out the file path which includes `*_test.go`, then execute go test.
```
HEAPSTER_ROOT=./
heapster::test-unit::find_dirs() {
  (
    cd ${HEAPSTER_ROOT}
    find . -not \( \
        \( \
          -path './_artifacts/*' \
          -o -path './docs/*' \
          -o -path './Godeps/*' \
          -o -path './deploy/*' \
          -o -path './version/*' \
          -o -path './vendor/*'\
        \) -prune \
      \) -name '*_test.go' -print0 | xargs -0n1 dirname | sed 's|^\./||' | sort -u
  )
}
set -- $(heapster::test-unit::find_dirs)
```
@piosz

**Comment by huangyuqi** [comment]
I will bring a fix PR with second way.

**Comment by piosz** [comment]
As a first step we can just get rid off `test-unit-cov`

**Comment by AlmogBaku** [comment]
@piosz @huangyuqi this can be solved by using glide #1314 (or specifically see my review comment in #1335 )

**Comment by huangyuqi** [comment]
@piosz I agree.  there are duplicate contents between `test-unit` and `test-unit-cov`.
